### ISSUE TITLE
Make default.nix an overlay

### ISSUE THREAD

**Issue by asymmetric** [dapphub/dapptools#317]
Similar to [mozilla-nixpkgs](https://github.com/mozilla/nixpkgs-mozilla).

Benefits:

* familiarity so that all one has to do is to [add our overlay](https://nixos.wiki/wiki/Overlays#Applying_overlays_manually).

I think for this to actually work well, we

**Comment by MrChico** [comment]
Wrt the haskell stuff I think we need to update to latest ghc, which shouldnt be too hard

**Comment by livnev** [comment]
@asymmetric Won't something like this work already?

```
  overlays = [ (import (builtins.fetchGit {
    url = "git@github.com:dapphub/dapptools.git";
    rev = "4b45ad07bafd6917f4bf5ba518d53f0ad1f82de3";
    } + /overlay.nix)
  ];
```

**Comment by asymmetric** [comment]
@livnev yeah absolutely, but it’s one more thing we have to explain people how to do (or they have to figure out themselves).

The main benefit would be standardization.

**Comment by livnev** [comment]
@asymmetric Agreed. 👍 

While we are on the subject, what is the advantage of the overlay approach versus doing something like 

```
dapptools = import (builtins.fetchGit {
    url = "git@github.com:dapphub/dapptools.git";
    rev = "4b45ad07bafd6917f4bf5ba518d53f0ad1f82de3";
    }) {};
```
and then using `dapptools.seth` etc.? That way we know we won't mess up people's haskell package set etc.

**Comment by asymmetric** [comment]
Now that I think of it, we would actually lose the binary cache, since everyone would be layering the overlay on top of a different version of nixpkgs. I think that's why we do it the current way in the first place.

The mozilla-nixpkgs overlay downloads pre-compiled binaries from mozilla's cache, so they don't have this problem.

:thinking:
